documentation/CAMARA-Security-Interoperability.md

```diff
@@ -0,0 +1,237 @@
+
+> Authentication Context Class Reference
+>>    OPs MUST support requests for specific Authentication Context Class Reference values via the acr_values parameter, as defined in Section 3.1.2. (Note that the minimum level of support required for this parameter is simply to have its use not result in an error.)
+
+OIDC also defines that the parameter acr_values is OPTIONAL and does not specify possible values. This leads to interoperability issues.
+
+This documents defines that Camara OpenId Providers MUST ignore the parameter acr_values. 
+
+This document defines that Camara Clients SHOULD not use the acr_values parameter. 
+
+> To foster interoperability a future version of this document might define values for the acr_values parameter acceptable in Camara.
+
+## Access Token Request
+
+This CAMARA document makes the scope parameter REQUIRED for the OAuth2 Client Credentials Grant.
+
+The client MUST authenticate with the authorization server as described in [Client Authentication Section](#client-authentication).
+
+## The Scope Parameter
+
+
+Scope values determine the specific CAMARA services being requested by the Service Provider, subject to the SP being registered to use those services. The scope values must be documented in the API OAS files by all Camara API subprojects. This document does not change OIDC definitions of scope values.
+
+
+---
+**NOTE**
+
+Scope values are an integral part of any OAuth2 and OIDC implementation. The RS enforces API access based on scope (if the Camara API subproject defines scopes).
+Therefore scopes should be available to API implementations.
+
+---
+
+
+## Missing "openid" scope
+
+[OIDC Core Authentication Request](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest) states the following about the value of scope.
+> REQUIRED. OpenID Connect requests MUST contain the openid scope value. If the openid scope value is not present, the behavior is entirely unspecified. Other scope values MAY be present. ...
+
+This document defines the following error handling for a missing "openid" value in scope. 
+Please refer to [Authentication Error Response](https://openid.net/specs/openid-connect-core-1_0.html#AuthError).
+
+If "openid" is missing in the scope value, the Authorization Server returns an HTTP response code of 400 (Bad Request) and an error invalid_request.
+
+## Purpose
+
+A transaction specific request parameter purpose as specified in [openid-connect-4-identity-assurance-1_0-13](https://openid.net/specs/openid-connect-4-identity-assurance-1_0.html#name-transaction-specific-purpos) MUST be used to allow a SP to state the purpose for the transfer of End-User data it is asking for.
+The purpose string MUST use below format for interoperability
+
+`dpv:<dpvValue>` 
+
+`<dpvValue>` is coming from [W3C DPV purpose definition](https://w3c.github.io/dpv/dpv/#vocab-purpose)
+
```

Comment:
```suggestion
## ID Token
Camara uses ID Token as defined in [OpenID Connect Core 1.0 incorporating errata set 2](https://openid.net/specs/openid-connect-core-1_0.html#CodeIDToken). As the specs describe, the ID Token's aud ( Audience) claim specifies who this ID Token is intended for and that it MUST contain the OAuth 2.0 client_id of the Relying Party as an audience value.
```

Reply:
I think it is good to reference the ID Token section in OIDC. However, for the part where it refers to the `aud` claim, it does not say anything different from what the standard already states, so I would not include it. If we start adding things like that, the document will end up looking more and more like a self-contained profile.

Reply:
@garciasolero Updated based on your comment.

Reply:
Please see https://github.com/AxelNennker/IdentityAndConsentManagement/blob/camara_oidc_profile/documentation/CAMARA-Security-Interoperability.md#id-token